soh/macosx/soh-macos.sh

```diff
@@ -65,5 +65,15 @@ while [ ! -e "$DATA_SHARE/oot.otr" ]; do
 		break
 done
 
-"$RESPATH"/soh-macos
+arch_name="$(uname -m)"
+launch_arch="arm64"
+if [ "${arch_name}" = "x86_64" ]; then
+    if [ "$(sysctl -in sysctl.proc_translated)" = "1" ]; then
+		launch_arch="arm64"
+	else
+		launch_arch="x86_64"
+	fi
+fi
```

Comment:
Would this also work?
```suggestion
if [ "${arch_name}" = "x86_64" ] && [ "$(sysctl -in sysctl.proc_translated)" = "0" ]; then
	launch_arch="x86_64"
fi
```